src/blog/2024/07/customise-theming-in-your-dashboards.md

```diff
@@ -0,0 +1,215 @@
+authors: ["sumit-shinde"]
+image: 
+tags:
+   - post
+   - dashboard
+---
+
+The latest release of Node-RED Dashboard 2.0 has taken customization to the next level. Previously, users enjoyed the flexibility of tweaking navigation sidebars, themes, and group and page padding. However, the header section remained static and non-customizable—until now. With the new update, you can fully personalize the header, adding unique elements to enhance your dashboard experience. In this article, we'll delve into these exciting new features, including theme adjustments, custom styling, and layout modifications, that empower you to tailor your Node-RED Dashboard like never before.
+
+<!--more-->
+
+## Adding Elements in the Header
+
+To add elements to the header, we utilize Vue's Teleport feature with ui-template widget, which allows elements to be seamlessly rendered in specific areas of the dashboard using CSS selectors such as IDs or classes. This method simplifies the process compared to manually positioning items with CSS, which can be complex, time-consuming, and potentially disruptive to other dashboard elements.
+
+### Left Side of the Header
+
+To render elements on the left side of the header, you can utilize the div element having the `#app-bar-title` ID, in which our page name is displayed.
+
+![Left Side Area](./images/left-side-area.png){data-zoomable}
+
+The `#app-bar-title` currently spans the full width of the header. When adding elements, they will occupy this area and align from left to right. However, if you add multiple elements in succession (using the default `display: flex`), they may extend beyond the width of the page view.
+
+#### Example: Adding Buttons
+
+1. Drag the ui-template widget onto the canvas.
+2. Double-click on it and select the scope to either `ui-scope` or `page-scope`. Selecting `ui-scope` will eliminate the need for multiple template widgets for the same purpose. For example, where you want the same buttons on all pages of your dashboard.
+3. Choose the page on which you want to render the buttons if you selected `page-scope`, or choose correct ui if `ui-scope` is selected.
+4. Paste the following Vue snippet into the template widget. In this snippet, note how we specify the "to" attribute targeting the `#app-bar-actions` ID in the teleport tag:
+
+```js
+<template>
+    <!-- Teleport the button to the #app-bar-actions area when mounted -->
+    <Teleport v-if="mounted" to="#app-bar-actions">
+        <v-btn>Button 1</v-btn>
+        <v-btn>Button 2</v-btn>
+        <v-btn>Button 3</v-btn>
+    </Teleport>
+</template>
+
+<script>
+    export default {
+        data() {
+            return {
+                mounted: false
+            }
+        },
+        mounted() {
+            // Set mounted to true when the component is mounted
+            this.mounted = true
+        }
+    }
+</script>
+```
+
+5. Next, you can customize further by adding more buttons or different elements inside the `<Teleport>` element.
```

Comment:
Missing step here where you need to turn off "Show Page Name in Navigation Bar" in the Dashboard settings, otherwise the buttons will show alongside the page name.